src/models/installation.ts

```diff
@@ -127,7 +147,43 @@ Installation.init({
 		type: DataTypes.INTEGER,
 		allowNull: true
 	}
-}, { sequelize });
+}, {
+	hooks: {
+		beforeSave: async (instance: Installation, opts) => {
+			if (opts.fields?.includes("sharedSecret")) {
+				if (!instance.encryptedSharedSecret) {
```

Comment:
no need to check for null.  If the sharedSecret changes, so does the encryptedSecret

Reply:
alright overwrite `encryptedSharedSecret`.  If you wanted to, you could remove the property from the class (but keep it in init), that way someone can't change the encrypted secret manually.  This will also work better long term as we do the switchover since there should be no reference to `encryptedSharedSecret` directly, only `sharedSecret`

Reply:
👍 I'll do this `always overwrite encryptedSharedSecret with the value of sharedSecret.`.

`you could remove the property from the class` <-- can't, the typings will complain (about no such property in the keyof this)

Reply:
yeah, but the great thing about typings is that you can ignore it if you want :)

In this special case, we're trying to not expose a property that shouldn't be accessed directly, so we do it covertly but need to break the typings for it to be saved using `setDataValue`

Reply:
Cool, let me try.

Reply:
ah right.  okay, then leave it in then.